SYCL/ESIMD/api/functional/ctors/common.hpp

```diff
@@ -29,54 +29,61 @@ using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 template <typename DataT>
 using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
 
-// Calls simd constructor in provided invocation context, which depends on the
-// TestCaseT type. TestCaseT is a struct, that should have call_simd_ctor method
-// that should return constructed object of simd class.
-// This function returns std::vector instance with the output data.
-template <typename DataT, int NumElems, typename TestCaseT>
-auto call_simd(sycl::queue &queue, const shared_vector<DataT> &ref_data) {
-
-  shared_vector<DataT> result{NumElems, shared_allocator<DataT>{queue}};
-
-  queue.submit([&](sycl::handler &cgh) {
-    const auto ref = ref_data.data();
-    auto out = result.data();
+template <typename DataT, int NumElems = 0>
```

Comment:
```suggestion
template <typename DataT, int NumElems>
```
That's a helper class specific for the `simd<T,N>` constructors, so looks like we don't need such special case. Or do we?

```suggestion
template <typename DataT, int NumElems, typename ContextT>
```
Moreover, looks like we are going to specify different contexts for every constructor covered, so we may move the `ContextT::get_description()` call out as well. What do you think?

Reply:
This comment was applied [here](https://github.com/intel/llvm-test-suite/pull/614/commits/f5fcba9f39f6f63e6e4c8d2de3b4ea8cc64a0444#diff-a1a30a06ba7da50ce8e13a9af9caec10d385910429792a1de956a7f15cc3d646R32)